cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -58,35 +60,56 @@ type ScaleSet struct {
 	minSize int
 	maxSize int
 
-	sizeMutex sync.Mutex
-	curSize   int64
-
+	enableForceDelete         bool
 	enableDynamicInstanceList bool
+	enableDetailedCSEMessage  bool
 
-	lastSizeRefresh   time.Time
+	// Current Size (Number of VMs)
+
+	// curSize tracks (and caches) the number of VMs in this ScaleSet.
+	// It is periodically updated from vmss.Sku.Capacity, with VMSS itself coming
+	// either from azure.Cache (which periodically does VMSS.List)
+	// or from direct VMSS.Get (always used for Spot).
+	curSize int64
+	// sizeRefreshPeriod is how often curSize is refreshed from vmss.Sku.Capacity.
+	// (Set from azureCache.refreshInterval = VmssCacheTTL or [defaultMetadataCache]refreshInterval = 1min)
 	sizeRefreshPeriod time.Duration
+	// lastSizeRefresh is the time curSize was last refreshed from vmss.Sku.Capacity.
+	// Together with sizeRefreshPeriod, it is used to determine if it is time to refresh curSize.
+	lastSizeRefresh time.Time
+	// getVmssSizeRefreshPeriod is how often curSize should be refreshed in case VMSS.Get call is used.
+	// (Set from GetVmssSizeRefreshPeriod, if specified = get-vmss-size-refresh-period = 30s
+	getVmssSizeRefreshPeriod time.Duration
+	// sizeMutex protects curSize (the number of VMs in the ScaleSet) from concurrent access
+	sizeMutex sync.Mutex
 
-	instancesRefreshPeriod time.Duration
-	instancesRefreshJitter int
+	InstanceCache
 
-	instanceMutex       sync.Mutex
-	instanceCache       []cloudprovider.Instance
-	lastInstanceRefresh time.Time
+	// uses Azure Dedicated Host
+	dedicatedHost bool
 }
 
 // NewScaleSet creates a new NewScaleSet.
-func NewScaleSet(spec *dynamic.NodeGroupSpec, az *AzureManager, curSize int64) (*ScaleSet, error) {
+func NewScaleSet(spec *dynamic.NodeGroupSpec, az *AzureManager, curSize int64, dedicatedHost bool) (*ScaleSet, error) {
 	scaleSet := &ScaleSet{
 		azureRef: azureRef{
 			Name: spec.Name,
 		},
-		minSize:                   spec.MinSize,
-		maxSize:                   spec.MaxSize,
-		manager:                   az,
-		curSize:                   curSize,
-		sizeRefreshPeriod:         az.azureCache.refreshInterval,
+
+		minSize: spec.MinSize,
+		maxSize: spec.MaxSize,
+
+		manager:           az,
+		curSize:           curSize,
+		sizeRefreshPeriod: az.azureCache.refreshInterval,
+		InstanceCache: InstanceCache{
+			instancesRefreshJitter: az.config.VmssVmsCacheJitter,
+		},
+
+		enableForceDelete:         az.config.EnableForceDelete,
```

Comment:
Not requiring this in master due to https://github.com/kubernetes/autoscaler/pull/6447